File: testapp/backend/controllers/LocalStore.js
const { Store } = require('./Store.js');
var fs = require('fs');

class LocalStore extends Store {

    constructor(filename) {
        super();
        this.filename = filename;
        fs.open(filename, 'r', function (fileExists, file) {
            if (fileExists) {
                fs.writeFile(filename, JSON.stringify({ data: [] }), (err) => {
                    if (err) console.error(err)
                });
            }
        });

    }

    postItem(item) {
        var f = JSON.parse(fs.readFileSync(this.filename).toString());
        f.data.push(item)
        fs.writeFileSync(this.filename, JSON.stringify(f));
    }

    getAllItems() {
        var f = JSON.parse(fs.readFileSync(this.filename).toString());
        return f.data
    }
}

module.exports.LocalStore = LocalStore;

File: testapp/backend/python/controllers/local_store.py
from .store import Store
from .item import Item
import json

class LocalStore(Store):
    file: str

    def __init__(self, file):
        self.file = file
    
    def postItem(self, item):
        with open(self.file,'r+') as file:
            data = json.load(file)
            data['data'].append(item.__dict__)
            file.seek(0)
            json.dump(data, file)
    
    def getAllItems(self):
        f = open(self.file)
        data = json.load(f)
        return data